Description: use timestamp ping only if Net::Ping version >= 2.50
 Avoid and error output if Net::Ping module is too old by just skipping the
 involved code.
Author: Guillaume Bougard <gbougard@teclib.com>
Bug: https://github.com/fusioninventory/fusioninventory-agent/issues/532
Last-Update: 2018-07-06

--- fusioninventory-agent-2.4.1.orig/lib/FusionInventory/Agent/Task/NetDiscovery.pm
+++ fusioninventory-agent-2.4.1/lib/FusionInventory/Agent/Task/NetDiscovery.pm
@@ -358,7 +358,7 @@ sub _scanAddress {
     my %device = (
         $INC{'Net/SNMP.pm'}      ? $self->_scanAddressBySNMP(%params)    : (),
         $INC{'Net/NBName.pm'}    ? $self->_scanAddressByNetbios(%params) : (),
-        $INC{'FusionInventory/Agent/Task/NetDiscovery/Ping.pm'} ?
+        $INC{'FusionInventory/Agent/Task/NetDiscovery/Ping.pm'} && ($Net::Ping::VERSION =~ /^([0-9.]*)/)[0] >= 2.50 ?
                                    $self->_scanAddressByTSPing(%params)  : (),
         $INC{'Net/Ping.pm'}      ? $self->_scanAddressByPing(%params)    : (),
         $params{arp}             ? $self->_scanAddressByArp(%params)     : (),
